Heimdallr/URLExtensions.swift

```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+@available(iOS 9.0, *)
+@available(OSXApplicationExtension 10.10, *)
+extension URL {
+
+    public var queryParameters: [String: String]? {
+        guard let components = URLComponents(url: self, resolvingAgainstBaseURL: true), let queryItems = components.queryItems else {
+            return nil
+        }
+
+        var parameters = [String: String]()
+        for item in queryItems {
```

Comment:
Isn't that a simple `map`?

Reply:
Not realy. Because `queryItems ` is an array and paramters is a dictionary.